src/app/ui/nav/Nav.tsx

```diff
@@ -18,12 +19,47 @@ const NAV_MENU = [
   },
 ];
 
+const NAV_ITEM_HEIGHT = 25;
+
 export default function Nav() {
+  const currentPagePath = window.location.pathname;
+
+  let activeIndex = 0;
+  NAV_MENU.forEach((item, index) => {
+    if (currentPagePath === item.link) {
+      activeIndex = index;
+    }
+  });
+
+  const [hovered, setHovered] = useState(activeIndex);
+
+  const isActive = (link: string): boolean => {
```

Comment:
nit: I like reserving 'is...' or 'are...' names for booleans
```suggestion
  const getIsActive = (link: string): boolean => {
```